src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -171,6 +172,20 @@ public AtomicInteger load(InetAddressAndPort inetAddress)
     private static final boolean disableSerialReadLinearizability =
         Boolean.parseBoolean(System.getProperty(DISABLE_SERIAL_READ_LINEARIZABILITY_KEY, "false"));
 
+    // allows custom implementation of mutateAtomically
+    private static final AtomicReference<AtomicMutator> atomicMutator = new AtomicReference<>();
```

Comment:
how about using `volatile` here and checking if `atomicMutator` is null in `StorageProxy#mutateAtomically`?

Reply:
this makes protection against mutator substitution post some mutation impossible, doesn't it? Or is there another way?